.github/workflows/pull-request.yml

```diff
@@ -23,3 +23,20 @@ jobs:
 
       - name: run yamllint
         run: yamllint .github/workflows/*.yml
+  checkpatch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
It looks like you don't use $ref.

Reply:
this is passed into checkout

Reply:
I think the reason I did this is otherwise it was giving me a merge which doesn't work for checkpatch

Reply:
Correct: by default `checkout` gives `pull/12355/merge` which is what most people want by default (even when they don't have a clue).


BTW: https://github.com/thesofproject/sof/blob/main/.github/workflows/codestyle.yml

Reply:
Interesting, ill look into that tomorrow, thanks for the pointer

Reply:
Nvm someone did it, any issue if we just use this? https://github.com/marketplace/actions/checkpatch-pl-pr-review

Reply:
Nvm someone did it, any issue if we just use this? https://github.com/marketplace/actions/checkpatch-pl-pr-review

Reply:
I just had a look at it. The code wouldn't pass `shellcheck` but it's short, understandable and uses the right API (i.e.: [NOT `git merge-base`](https://github.com/actions/checkout/issues/552#issuecomment-1344881907)

Worth a try!